include/ck/utility/data_type.hpp

```diff
@@ -7,12 +7,28 @@
 
 namespace ck {
 
+template <class... T>
+__host__ __device__ void print_type()
+{
+    static_assert(sizeof...(T) < 0, "Printing type T.");
+}
+
+template <class... T>
+__host__ __device__ void print_type(T&&...)
+{
+    static_assert(sizeof...(T) < 0, "Printing type T.");
+}
+
 using bhalf_t = ushort;
 using half_t  = _Float16;
 using int4_t  = _BitInt(4);
 using f8_t    = _BitInt(8);
 using bf8_t   = unsigned _BitInt(8);
 
+// this is not f4_t x 2 as a vector type, but 2 fp4_t packed in one byte, sub-byte access with
+// vector is not allowed. using packed_f4_t  = _BitInt(8);
+using packed_f4_t = unsigned char; // This the scalar type and must not be used
```

Comment:
@zjing14 Do you have any idea on how can we make distinct type from _BitInt(8)?

f8_t is already _BitInt(8), if we reuse _BitInt(8) again, is_same_v(f8_t, packed_f4_t) will be true. In the future, i'd assume there will be f8_e4m3fnuz_t, f8_e4m3fn_t and f8_e5m2_t, and they should all be nominally different type.

Reply:
@cloudhan, we are currently working on refactoring data types to be struct-based types instead of _BitInt types. It will allow us to add as many data types as we want and avoid ""is_same_v(f8_t, packed_f4_t)" ambiguities.